test/jdk/java/foreign/trivial/TestTrivial.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @enablePreview
+ * @library ../ /test/lib
+ * @requires ((os.arch == "amd64" | os.arch == "x86_64") & sun.arch.data.model == "64") | os.arch == "aarch64"
+ * @run testng/othervm --enable-native-access=ALL-UNNAMED TestTrivial
+ */
+
+import org.testng.annotations.Test;
+
+import java.lang.foreign.Arena;
+import java.lang.foreign.FunctionDescriptor;
+import java.lang.foreign.Linker;
+import java.lang.foreign.MemoryLayout;
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.SegmentAllocator;
+import java.lang.foreign.StructLayout;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.VarHandle;
+
+import static org.testng.Assert.assertEquals;
+
+public class TestTrivial extends NativeTestHelper {
```

Comment:
Should we perhaps have another subclass of `TestDowncallBase` to test all combinations - just in case?

Reply:
Seems like overkill, tbh, since `trivial` doesn't affect any of the classification logic. I'm already testing all the cases that are handled differently by the downcall stub (where `trivial` does matter).

Also, ultimately the stuff covered by TestDowncall is limited (no unions, nested structs, array members. Limited set of primitive types). I think long term I'd rather implement some kind of fuzzing-based test, where we generate an arbitrary FunctionDescriptor/linker options/C function, and then try to link and call that. But this requires generating and compiling C code on the fly during the tests (which I expect will be problematic to get working).

Reply:
I could maybe add a few more test cases, to the tune of what we do for the CaptureCallState tests. But my feeling is that if we go to the level of TestDowncall, we don't really get any more coverage from that.